lib/msal-node/src/network/HttpClient.ts

```diff
@@ -274,3 +281,45 @@ const networkRequestViaHttps = <T>(
     });
 };
 
+/**
+ * Check if extra parsing is needed on the repsonse from the server
+ * @param statusCode {number} the status code of the response from the server
+ * @param statusMessage {string | undefined} the status message of the response from the server
+ * @param headers {Record<string, string>} the headers of the response from the server
+ * @param body {string} the body from the response of the server
+ * @returns {Object} JSON parsed body or error object
+ */
+const parseBody = (statusCode: number, statusMessage: string | undefined, headers: Record<string, string>, body: string) => {
+    /*
+     * Informational responses (100 – 199)
+     * Successful responses (200 – 299)
+     * Redirection messages (300 – 399)
+     * Client error responses (400 – 499)
+     * Server error responses (500 – 599)
+     */
+    
+    let parsedBody;
+    try {
+        parsedBody = JSON.parse(body);
+    } catch (error) {
+        let errorType;
+        let errorDescriptionHelper;
+        if ((statusCode >= HttpStatus.CLIENT_ERROR_RANGE_START) && (statusCode <= HttpStatus.CLIENT_ERROR_RANGE_END)) {
```

Comment:
Here is where I think we should address `HttpStatus.REDIRECT_ERROR_RANGE_START` && .... to classify it as a certain type of error.

Reply:
@bgavrilMS what are your thoughts on this? After some research, it looks like redirections (http status 300-399) can be sent without a body, which would trigger this error. However, redirections aren't errors. I don't think the response body would be used in the case of a redirection anyways... should we just send back an empty one in the case of a redirect?

Reply:
I doubt that your HttpClient implementation handles redirects :( But the /token endpoint is not expected to reply with 3xx, so we should be fine.